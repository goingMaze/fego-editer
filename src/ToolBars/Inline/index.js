import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { RichUtils } from 'draft-js'
import { ToolBarBtn } from 'utils'
import Iconfont from 'iconfont'

const INLINE_TYPES = [
	{ label: 'bold', style: 'BOLD', icon: 'iconBold', title: '加粗' },
	{ label: 'italic', style: 'ITALIC', icon: 'iconItalic', title: '斜体' },
	{ label: 'underline', style: 'UNDERLINE', icon: 'iconUnderline', title: '下划线' },
];

export default class FontSize extends Component {
	static propTypes = {
		editorState: PropTypes.object.isRequired,
		onChange: PropTypes.func.isRequired,
		config: PropTypes.array.isRequired,
		modalManage: PropTypes.object.isRequired
	}
	onToggle = inlineStyle => {
		let { editorState, onChange } = this.props;
		onChange(RichUtils.toggleInlineStyle(editorState, inlineStyle))
	}
	render() {
		let { editorState, config } = this.props;
		let currentStyle = editorState.getCurrentInlineStyle();
		return (
			<div className="FegoEditor-toolbars">
				{
					INLINE_TYPES.filter(item => config.includes(item.label)).map(type =>
						<ToolBarBtn
							key={type.label} title={type.title}
							active={currentStyle.has(type.style)}
							onToggle={this.onToggle}
							style={type.style}
						>
							{Iconfont[type.icon]}
						</ToolBarBtn>
					)
				}
			</div>
		)
	}
}