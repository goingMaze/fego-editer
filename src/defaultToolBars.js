const defaultToolBars = {
	options: ['BlockType', 'Color', 'FontFamily', 'FontSize', 'Image', 'Inline', 'Link', 'List', 'TextAlign', 'Video', 'History'],
	BgColor: [],
	BlockType: ['正文', 'H1', 'H2', 'H3', 'H4', 'H5', 'H6'],
	Color: ['rgb(97,189,109)', 'rgb(26,188,156)', 'rgb(84,172,210)', 'rgb(44,130,201)',
		'rgb(147,101,184)', 'rgb(71,85,119)', 'rgb(204,204,204)', 'rgb(65,168,95)', 'rgb(0,168,133)',
		'rgb(61,142,185)', 'rgb(41,105,176)', 'rgb(85,57,130)', 'rgb(40,50,78)', 'rgb(0,0,0)',
		'rgb(247,218,100)', 'rgb(251,160,38)', 'rgb(235,107,86)', 'rgb(226,80,65)', 'rgb(163,143,132)',
		'rgb(239,239,239)', 'rgb(255,255,255)', 'rgb(250,197,28)', 'rgb(243,121,52)', 'rgb(184,49,47)',
		'rgb(209,72,65)', 'rgb(124,112,107)', 'rgb(209,213,216)'],
	FontFamily: [{
		name: 'SimHei', family: '"SimHei", "黑体"'
	}, {
		name: 'Yahei', family: '"Microsoft Yahei", "微软雅黑"'
	}, {
		name: 'KaiTi', family: '"KaiTi", "楷体"'
	}, {
		name: 'FangSong', family: '"FangSong", "仿宋"'
	}],
	FontSize: [16, 20, 30, 40, 50, 70, 100, 200],
	History: ['undo', 'redo'],
	Image: {
		uploadUrl: '',
		uploadCallback: null
	},
	Inline: ['bold', 'italic', 'underline'],
	Link: {},
	List: ['ul', 'ol', 'indent', 'outdent'],
	TextAlign: ['left', 'center', 'right', 'justify'],
	Video: {},
	Remove: {}
};

export default defaultToolBars
