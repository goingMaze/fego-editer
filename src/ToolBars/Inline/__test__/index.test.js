import React from 'react'
import { mount, render } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { EditorState, ContentState, convertFromHTML } from 'draft-js'
import Inline from '..'
import defaultToolBars from '../../../default'
import ModalManage from '../../../utils/ModalManage'

describe('Inline', () => {
	const contentState = ContentState.createFromBlockArray(convertFromHTML('<div>test</div>'))
	const editorState = EditorState.createWithContent(contentState)
	const wrapper = mount(
		<Inline editorState={editorState} onChange={() => { }}
			config={defaultToolBars.Inline} modalManage={new ModalManage()}
		/>
	);
	it('渲染时开头应该有一个div', () => {
		expect(wrapper.html().startsWith('<div')).toEqual(true)
	})
	it('应该有3个toolBarBtn子组件', () => {
		expect(wrapper.find('ToolBarBtn').length).toBe(3)
	})
	it('config应该为默认值', () => {
		expect(wrapper.prop('config')).toEqual(defaultToolBars.Inline)
	})
	it('快照', () => {
		const rendered = render(
			<Inline editorState={editorState} onChange={() => { }}
				config={defaultToolBars.Inline} modalManage={new ModalManage()}
			/>
		)
		expect(toJSON(rendered)).toMatchSnapshot()
	})
})