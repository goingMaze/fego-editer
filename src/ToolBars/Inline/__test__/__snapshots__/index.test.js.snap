// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inline 快照 1`] = `
<div
  class="FegoEditor-toolbars"
>
  <div
    class="FegoEditor-toolBarBtn "
    title="加粗"
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M524.8 819.2 256 819.2 256 153.6 510.8 153.6C632.133939 157.8873 697.466619 211.477746 706.8 314.372947 704.933325 397.975297 666.667039 449.422126 592 468.714976L592 471.930435C677.867095 489.079635 724.533294 541.598275 732 629.487923 732 755.963274 662.934026 819.2 524.8 819.2L524.8 819.2ZM505.2 520.162319 354 520.162319 354 735.59807 508 735.59807C586.400394 735.59807 625.6 699.156562 625.6 626.272466 619.999974 557.675663 579.867039 522.305971 505.2 520.162319L505.2 520.162319ZM491.2 240.417391 354 240.417391 354 433.344928 491.2 433.344928C564.000364 431.201277 601.333325 397.975201 603.2 333.6657 601.333325 271.49985 564.000364 240.417391 491.2 240.417391L491.2 240.417391Z"
      />
    </svg>
  </div>
  <div
    class="FegoEditor-toolBarBtn "
    title="斜体"
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M776.209986 181.163059l0 57.898665-88.015589 0-220.039485 539.282367 88.016613 0 0 58.328453-308.056098 0 0-58.328453 88.016613 0 220.039485-539.282367-88.016613 0 0-57.898665L776.209986 181.163059z"
      />
    </svg>
  </div>
  <div
    class="FegoEditor-toolBarBtn "
    title="下划线"
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M512 725.333c141.227 0 256-114.773 256-256V128H661.333v341.333c0 82.347-66.986 149.334-149.333 149.334S362.667 551.68 362.667 469.333V128H256v341.333c0 141.227 114.773 256 256 256z m-298.667 85.334V896h597.334v-85.333H213.333z"
        fill=""
      />
    </svg>
  </div>
</div>
`;

exports[`Inline 快照 1`] = `
<div
  class="FegoEditor-toolbars"
>
  <div
    class="FegoEditor-toolBarBtn "
    title="加粗"
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M524.8 819.2 256 819.2 256 153.6 510.8 153.6C632.133939 157.8873 697.466619 211.477746 706.8 314.372947 704.933325 397.975297 666.667039 449.422126 592 468.714976L592 471.930435C677.867095 489.079635 724.533294 541.598275 732 629.487923 732 755.963274 662.934026 819.2 524.8 819.2L524.8 819.2ZM505.2 520.162319 354 520.162319 354 735.59807 508 735.59807C586.400394 735.59807 625.6 699.156562 625.6 626.272466 619.999974 557.675663 579.867039 522.305971 505.2 520.162319L505.2 520.162319ZM491.2 240.417391 354 240.417391 354 433.344928 491.2 433.344928C564.000364 431.201277 601.333325 397.975201 603.2 333.6657 601.333325 271.49985 564.000364 240.417391 491.2 240.417391L491.2 240.417391Z"
      />
    </svg>
  </div>
  <div
    class="FegoEditor-toolBarBtn "
    title="斜体"
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M776.209986 181.163059l0 57.898665-88.015589 0-220.039485 539.282367 88.016613 0 0 58.328453-308.056098 0 0-58.328453 88.016613 0 220.039485-539.282367-88.016613 0 0-57.898665L776.209986 181.163059z"
      />
    </svg>
  </div>
  <div
    class="FegoEditor-toolBarBtn "
    title="下划线"
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M512 725.333c141.227 0 256-114.773 256-256V128H661.333v341.333c0 82.347-66.986 149.334-149.333 149.334S362.667 551.68 362.667 469.333V128H256v341.333c0 141.227 114.773 256 256 256z m-298.667 85.334V896h597.334v-85.333H213.333z"
        fill=""
      />
    </svg>
  </div>
</div>
`;
